### PR TITLE
feat(context-menu): added context menu (#2865)

### PR DESCRIPTION
**Pull request by lskramarov**

### CODE REVIEW

File: packages/primitives/context-menu/src/context-menu-content.directive.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    readonly highlighted = new Subject<RdxContextMenuItemDirective>();
+    readonly menuTrigger = inject(RdxContextMenuTriggerDirective, { optional: true });
+
+    @Input() onEscapeKeyDown: (event?: Event) => void = () => undefined;
+    @Input() closeOnEscape = true;
+
+    constructor() {
+        super();
+
+        this.highlighted.pipe(startWith(null), pairwise()).subscribe(([prev, item]) => {
```
Comment (artembelik) [inline packages/primitives/context-menu/src/context-menu-content.directive.ts:32]:
```suggestion
        this.highlighted.pipe(startWith(null), pairwise(), takeUntilDestroyed()).subscribe(([prev, item]) => {
```

File: packages/primitives/context-menu/src/context-menu-item-radio-group.directive.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    host: {
+        role: 'group'
+    },
+    providers: [{ provide: UniqueSelectionDispatcher, useClass: UniqueSelectionDispatcher }]
+})
+export class RdxContextMenuItemRadioGroupDirective<T> implements AfterContentInit {
+    private readonly selectionDispatcher = inject(UniqueSelectionDispatcher);
+
+    @Input()
+    set value(id: T | null) {
```
Comment (artembelik) [inline packages/primitives/context-menu/src/context-menu-item-radio-group.directive.ts:16]:
```suggestion
    set value(value: T | null) {
```

File: packages/primitives/context-menu/src/context-menu-item-radio-group.directive.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    get value(): T | null {
+        return this._value;
+    }
+
+    private _value: T | null = null;
+
+    @Output() readonly valueChange = new EventEmitter();
+
+    ngAfterContentInit(): void {
+        this.selectionDispatcher.notify(this.value as string, '');
```
Comment (artembelik) [inline packages/primitives/context-menu/src/context-menu-item-radio-group.directive.ts:29]: maybe?
```
if (typeof this.value === "string") {
   this.selectionDispatcher.notify(...);
}
```

**Review by artembelik — APPROVED** [review]

**Review by pimenovoleg — APPROVED** [review]

File: packages/primitives/context-menu/src/context-menu-item-radio-group.directive.ts (hunk 3 above)
Comment (lskramarov) [reply packages/primitives/context-menu/src/context-menu-item-radio-group.directive.ts:29]: no, value has generic type

Comment (artembelik) [reply packages/primitives/context-menu/src/context-menu-item-radio-group.directive.ts:29]: but `selectionDispatcher.notify` waiting for a string as first arg
potential for error

Comment (lskramarov) [reply packages/primitives/context-menu/src/context-menu-item-radio-group.directive.ts:29]: this is a problem of UniqueSelectionDispatcher :) It compares two values, and it doesn't matter what type they are. I don't know why UniqueSelectionDispatcher is written without generic types... maybe in the future we will write our own UniqueSelectionDispatcher with generic types

**Pull request merged**